packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx

```diff
@@ -113,6 +123,15 @@ const VisitHeader: React.FC = () => {
   const navMenuItems = useAssignedExtensions('patient-chart-dashboard-slot').map((extension) => extension.id);
   const { startVisitLabel } = useConfig();
 
+  const store = getWorkspaceStore();
+  const state = store.getState();
+  const workSpaceProps: workspaceProps = state.openWorkspaces?.[0] || {};
+
+  const formWorkspace = state.openWorkspaces?.find((workspace) => workspace.type === 'form');
+  const orderWorkspace = state.openWorkspaces?.find((workspace) => workspace.type === 'order');
+  const visitNoteWorkspace = state.openWorkspaces?.find((workspace) => workspace.type === 'visit-note');
+  const hasOpenForm = formWorkspace || orderWorkspace || visitNoteWorkspace;
```

Comment:
So, my suggestion was really about generifying things here. Keep in mind that the workspace if pluggable. We do not really know before hand the list of possible workspaces. My suggestion was that we should be able to iterate through the list of workspaces and just call the `closeWorkspace()` on each.

Reply:
I am sorry @ibacher , I do not fully understand the above, do you mean call the `promptBeforeClosing()` or `closeWorskpace()`